source/AI.cpp

```diff
@@ -89,12 +89,16 @@ namespace {
 	// ship. Carried escorts are waited for in AI::Step.
 	bool EscortsReadyToJump(const Ship &ship)
 	{
-		for(const weak_ptr<Ship> &escort : ship.GetEscorts())
+		bool shipIsYours = ship.IsYours();
+		for(const weak_ptr<Ship> &ptr : ship.GetEscorts())
 		{
-			shared_ptr<const Ship> locked = escort.lock();
-			if(locked && !locked->IsDisabled() && !locked->CanBeCarried()
-					&& locked->GetSystem() == ship.GetSystem()
-					&& locked->JumpFuel() && !locked->IsReadyToJump(true))
+			shared_ptr<const Ship> escort = ptr.lock();
+			// Skip escorts which are not player-owned and not escort mission NPCs.
+			if(!escort || (shipIsYours && !escort->IsYours() && !escort->GetPersonality().IsEscort()))
```

Comment:
You also need to check the hostility of NPCs. It is perfectly valid to assign `personality escort` to a hostile NPC.

Reply:
As in check if NPC government is hostile against player government?

Reply:
@tehhowch see d397ff89db0ebacf6d198eef3c0f285ca8c119d5; I think I have addressed this feedback.